// @inlang/sdk-js@0.11.8/dist/adapter-sveltekit/ast-transforms/test.utils.js
import { deepmerge } from "deepmerge-ts";
export const initTransformConfig = (overrides = {}) => deepmerge({
    isStatic: false,
    languageInUrl: false,
    cwdFolderPath: "",
    rootRoutesFolder: "",
    sourceFileName: "",
    sourceMapName: "",
    inlang: {
        referenceLanguage: "en",
        languages: ["en", "de"],
        readResources: async () => [],
        writeResources: async () => undefined,
        sdk: {
            debug: false,
            languageNegotiation: {
                strict: false,
                strategies: [{ type: "localStorage", key: "language" }],
            },
            resources: {
                cache: "build-time",
            },
            routing: {
                exclude: [],
            },
        },
    },
    svelteKit: {
        usesTypeScript: false,
        version: undefined,
        files: {
            appTemplate: "src/app.html",
            routes: "src/routes",
            serverHooks: "src/hooks.server",
        },
    },
}, overrides);
